### DISCUSSION TITLE
License Related Help

### DISCUSSION THREAD

**Discussion by Matthew-H-CT** [community/community#22724] [New to GitHub]
<p>On <a href="https://choosealicense.com/licenses/gpl-2.0/" rel="nofollow noopener">https://choosealicense.com/licenses/gpl-2.0/</a> in the optional steps section it talks about " If you would like your project to adopt the GPL-3.0’s cure provision, add the <a href="https://github.com/gplcc/gplcc/blob/22e11a340f9f502d5a5af63bbda7ae656fa40894/Project/COMMITMENT" rel="nofollow noopener">text</a> of the <a href="https://gplcc.github.io/gplcc/" rel="nofollow noopener">GPL Cooperation Commitment</a> to a file named <code>COMMITMENT</code> in the same directory as your GPL-2.0 license file." and I was wondering what would change if I added/had the Commitment compared to not adding/having it?</p>

**Comment by lee-dohm** [comment]
<p>Hello, thanks for joining us!</p>
<p>While we have created the <a href="https://choosealicense.com">https://choosealicense.com</a> site to give a broad overview of the choices one has in licensing a project, we are not able to give legal advice. If you have questions about what licensing agreements mean or obligate you to, we suggest that you contact your own legal counsel.</p>
<p>Thanks again for being here.</p>

**Comment by Matthew-H-CT** [comment]
<p>But wont a legal counsel cost money I dont really wanna pay for a simple thing like this all I wanna know is what this changes I already know what gpl v2 is but not what the commitment changes.</p>

**Comment by lee-dohm** [comment]
<aside class="quote no-group" data-username="Matthew-H-CT" data-post="3" data-topic="125469">
<div class="title">
<div class="quote-controls"></div>
 Matthew-H-CT:</div>
<blockquote>
<p>I dont really wanna pay for a simple thing like this all I wanna know is what this changes</p>
</blockquote>
</aside>
<p>Regardless of how simple the request may seem, we’re neither qualified to nor allowed to give legal advice of any kind. And yes, giving you an interpretation of what something means legally definitely qualifies as “legal advice”.</p>
<p>I understand that this is disappointing. Unfortunately, our policy is quite strict on this point.</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="Matthew-H-CT" data-post="1" data-topic="125469">
<div class="title">
<div class="quote-controls"></div>
 Matthew-H-CT:</div>
<blockquote>
<p>what would change if I added/had the Commitment compared to not adding/having it?</p>
</blockquote>
</aside>
<p>You can read an introductory text about what the GPL Cooperation Commitment is at the link you provided:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://gplcc.github.io/gplcc/" target="_blank" rel="nofollow noopener">GPL Cooperation Commitment</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://gplcc.github.io/gplcc/" target="_blank" rel="nofollow noopener">Join the GPL Cooperation Commitment</a></h3>

<p>Keep open source licenses fair and predictable</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>but you should always bare in mind that the actual <em><strong>legal text</strong></em> of the license is what ultimately matters (i.e. if you end up in a dispute in court):</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/gplcc/gplcc/blob/22e11a340f9f502d5a5af63bbda7ae656fa40894/Project/COMMITMENT" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/gplcc/gplcc/blob/22e11a340f9f502d5a5af63bbda7ae656fa40894/Project/COMMITMENT" target="_blank" rel="nofollow noopener">gplcc/gplcc/blob/22e11a340f9f502d5a5af63bbda7ae656fa40894/Project/COMMITMENT</a></h4>
<pre><code class="lang-">GPL Cooperation Commitment
Version 1.0

Before filing or continuing to prosecute any legal proceeding or claim
(other than a Defensive Action) arising from termination of a Covered
License, we commit to extend to the person or entity ('you') accused
of violating the Covered License the following provisions regarding
cure and reinstatement, taken from GPL version 3. As used here, the
term 'this License' refers to the specific Covered License being
enforced.

    However, if you cease all violation of this License, then your
    license from a particular copyright holder is reinstated (a)
    provisionally, unless and until the copyright holder explicitly
    and finally terminates your license, and (b) permanently, if the
    copyright holder fails to notify you of the violation by some
    reasonable means prior to 60 days after the cessation.

    Moreover, your license from a particular copyright holder is
    reinstated permanently if the copyright holder notifies you of the
</code></pre>

  This file has been truncated. <a href="https://github.com/gplcc/gplcc/blob/22e11a340f9f502d5a5af63bbda7ae656fa40894/Project/COMMITMENT" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p><a class="mention" href="/lee-dohm">@lee-dohm</a> is right about the fact that explaining a license amounts to legal advise. You’ll need to consult a lawyer for various reasons, the most obvious one being that he/she will be professionally qualified to do so, and with your best interests in view, but also because licenses are indeed subject to legal interpretations according to the laws of the country — which means that the laws of different countries might have an impact on how a license might be interpreted in the court of a given country, due to the local context (which is why many EULAs explicitly state in which country any legal dispute will be settled, in case of breaches of contract).</p>
<p><strong>EDIT</strong> — FYI <a class="mention" href="/matthew-h-ct">@Matthew-H-CT</a>, you might want to ask license-related questions on <strong>Open Source Stack Exchange</strong> or on <strong>Law Stack Exchange</strong>, two dedicated forums on legal matters, where you’ll find legal experts (from various countries) willing to discuss similar legal topics:</p>
<ul>
<li><a href="https://opensource.stackexchange.com/" rel="nofollow noopener">https://opensource.stackexchange.com/</a></li>
<li><a href="https://law.stackexchange.com/" rel="nofollow noopener">https://law.stackexchange.com/</a></li>
</ul>

**Comment by Matthew-H-CT** [comment]
<p>thanks so much for provideing those links they helped me a lot.</p>
